variable/test/comparison_test.cpp

```diff
@@ -53,7 +54,17 @@ TYPED_TEST(IsCloseTest, rtol_when_variables_outside_tolerance) {
   const auto atol = makeVariable<TypeParam>(Values{0});
   EXPECT_EQ(isclose(a, b, rtol, atol), false * units::one);
 }
-
+TEST(IsCloseTest, with_vectors) {
+  const auto a =
+      makeVariable<Eigen::Vector3d>(Values{Eigen::Vector3d{1, 1, 1}});
+  const auto b =
+      makeVariable<Eigen::Vector3d>(Values{Eigen::Vector3d{1, 1, 1}});
+  const auto rtol = 0.0 * units::one;
+  const auto atol = 1.0 * units::one;
+  EXPECT_EQ(isclose(a, b, rtol, atol),
```

Comment:
This highlights a problem with outputs that isn't ideal. 

We could rectify by internally doing the equivalent of this:
```sc.all(dim='u', x=sc.fold(x, 'x', {'v': int(sum(x.shape)/3), 'u': 3}))```

Reply:
I think you should use something like this in `isclose`:
```cpp
return all(isclose(a.elements<Eigen::Vector3d>(),
                   b.elements<Eigen::Vector3d>(), rtol, atol, equal_nans),
           Dim::InternalStructureComponent);
```

Reply:
Yes that might do it!